invisible_cities/reco/paolina_functions.py

```diff
@@ -321,38 +321,32 @@ def drop_voxel(voxels: Sequence[Voxel], the_vox: Voxel) -> int:
         qs  = [getattr(h, e_type) for h in the_vox.hits]
 
         if len(pos) == 0:
-            min_dist = 1e+06
-            min_v = voxels[0]
-            for v in voxels:
-                if neighbours(the_vox, v):
-                    dist = np.linalg.norm(the_vox.pos - v.pos)
-                    if dist < min_dist:
-                        min_dist = dist
-                        min_v = v
-
-            ### add voxel energy to voxel
-            min_v.E += the_vox.E
+            min_dist  = min([np.linalg.norm(the_vox.pos-v.pos) for v in voxels  if neighbours(the_vox, v)])
+            min_v     = [v for v in voxels  if (neighbours(the_vox, v) and np.isclose(np.linalg.norm(the_vox.pos-v.pos), min_dist))]
+
```

Comment:
I think you can first define `neighbour_voxels = [v for v in voxels  if neighbours(the_vox, v)]` and then use it for `min_dist` and for `min_v`. This would reduce looping twice over the voxels that you already know they are not close.

Reply:
Also maybe you can define `def dist_to_the_vox(vox): return np.linalg(the_vox.pos - vox.pos)` to improve the readability of the code

Reply:
Discarded because the author doesn't think it adds anything to the code